src/message.cpp

```diff
@@ -37,9 +39,20 @@ static QCString        g_warnlogFile;
 static bool            g_warnlogTemp = false;
 static std::atomic_bool g_warnStat = false;
 static std::mutex      g_mutex;
+static std::unordered_set<std::string> g_warnHash;
 
 //-----------------------------------------------------------------------------------------
 
+static bool checkWarnMessage(QCString result)
+{
+  uint8_t md5_sig[16];
+  char sigStr[33];
+  MD5Buffer(result.data(),result.length(),md5_sig);
+  MD5SigToString(md5_sig,sigStr);
+
+  return g_warnHash.insert(sigStr).second;
```

Comment:
This is not thread-safe. I think it will lead to memory corruption if multiple threads are enabled. Should only be called when the `g_mutex` lock is held.

Reply:
I agree, I thought that I checked whether or not the mutex was held, will have to recheck.

Reply:
Fixed 2 place, but wouldn't the problem also not have been in the original code or is `fmt::print(g_warnFile,"{}{}",g_warningStr,vformat(fmt,args));` thread safe?